kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -255,17 +233,19 @@ private void setState(State state) {
     this.state = state;
   }
 
-  private boolean shouldRotate(Long timestamp) {
-    boolean scheduledRotation = rotateScheduleIntervalMs > 0
-                                    && timestamp != null
-                                    && timestamp >= nextScheduledRotate;
-    boolean messageSizeRotation = recordCount >= flushSize;
-
-    log.trace("Should rotate (count {} >= flush size {} and schedule interval {} next schedule {} timestamp {})? {}",
-              recordCount, flushSize, rotateScheduleIntervalMs, nextScheduledRotate, timestamp,
-              scheduledRotation || messageSizeRotation);
+  private boolean rotateOnTime(String encodedPartition) {
+    boolean timeBasedRotation =
+        rotateIntervalMs > 0 && recordCount > 0 && !encodedPartition.equals(currentEncodedPartition);
```

Comment:
Messages are not guaranteed to arrive ordered with respect to the timestamp (especially with `CreateTime` timestamps), which might result in lots of small files being created around the partition boundary. Or am I missing something?